internal/backend/remote-state/s3/client.go

```diff
@@ -308,10 +329,46 @@ func (c *RemoteClient) Lock(info *statemgr.LockInfo) (string, error) {
 			Err:  err,
 			Info: lockInfo,
 		}
-		return "", lockErr
+		return lockErr
 	}
 
-	return info.ID, nil
+	return nil
+}
+
+// s3Lock expects the statemgr.LockInfo#ID to be filled already
+func (c *RemoteClient) s3Lock(info *statemgr.LockInfo) error {
+	if !c.useLockfile {
+		return nil
+	}
+	info.Path = c.lockPath()
+
+	lInfo := info.Marshal()
+	putParams := &s3.PutObjectInput{
+		ContentType:   aws.String(contentTypeJSON),
+		ContentLength: aws.Int64(int64(len(lInfo))),
+		Bucket:        aws.String(c.bucketName),
+		Key:           aws.String(c.lockFilePath()),
+		Body:          bytes.NewReader(lInfo),
+		IfNoneMatch:   aws.String("*"),
+	}
+
+	ctx := context.TODO()
+	ctx, _ = attachLoggerToContext(ctx)
+	_, err := c.s3Client.PutObject(ctx, putParams)
+	if err != nil {
```

Comment:
Is there a particular type of error we should be checking for here?  I see that this follows the same pattern that exists in the dynamo scenario.

Reply:
Looking at this again, I'm happy keeping it as is.  Simpler and consistent is probably better.